sycl/source/detail/platform_impl.cpp

```diff
@@ -389,6 +390,31 @@ bool platform_impl::has(aspect Aspect) const {
 #include <CL/sycl/info/platform_traits.def>
 #undef __SYCL_PARAM_TRAITS_SPEC
 
+context platform_impl::getDefaultContext() {
+  if (!MDefaultContexts.empty())
```

Comment:
It seems it can be a source of races. For example: one thread enters the if while another thread removes a context in the `popDefaultContext` method. Could you please fix that?